akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentActorContext.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import akka.actor.typed.internal.adapter.ActorContextAdapter
+import akka.actor.typed.scaladsl.ActorContext
+import akka.persistence.typed.internal.Running
+
+object PersistentActorContext {
+
+  /**
+   * Access the to details about the persistent actor from the actor context, can only be used in response
+   * to a command, will throw `IllegalState` for other cases
```

Comment:
it might be needed in more places than receiving command, such as onRecoveryCompleted, but that should be doable by implementing WithStateAccessible in more places if needed